<h1>Question 10</h1>

In [27]:
import numpy as np
import pandas as pd
df = pd.DataFrame(np.random.uniform(1, 2, (20, 5)), columns=list('abcde'))
df

a         b         c         d         e
0   1.880843  1.320479  1.416674  1.005828  1.179477
1   1.650108  1.989020  1.484707  1.523982  1.247570
2   1.455549  1.440511  1.403900  1.473522  1.587379
3   1.851480  1.750142  1.445667  1.253328  1.573210
4   1.922452  1.898862  1.697920  1.710930  1.426042
5   1.979674  1.068126  1.353029  1.726961  1.659205
6   1.507097  1.133749  1.734548  1.663728  1.535361
7   1.274441  1.972587  1.701627  1.220980  1.003133
8   1.690039  1.568925  1.929697  1.851065  1.215951
9   1.474857  1.052607  1.276691  1.635052  1.416964
10  1.466008  1.320530  1.511068  1.668908  1.229427
11  1.751057  1.475668  1.581855  1.179564  1.427405
12  1.865198  1.139561  1.733581  1.905747  1.603171
13  1.732365  1.542237  1.128564  1.171042  1.027721
14  1.646544  1.799957  1.884981  1.297664  1.532391
15  1.941233  1.687333  1.459401  1.433500  1.925718
16  1.964113  1.042838  1.978120  1.104549  1.778962
17  1.317641  1.323893  1.681040  1.645931  1.634549
18  1.362270  1.980773  1.722718  1.355916  1.972880
19  1.269388  1.749474  1.739397  1.235419  1.448450

In [28]:
print(max(df.std()))
print(df.std())
print(df.std().idxmax())

0.3296828505691937
a    0.243562
b    0.329683
c    0.224060
d    0.263827
e    0.266171
dtype: float64
b


Note: this answer would be change after rerunning the program
Column with the highest standard deviation is column b with standard deviation=0.329683

In [29]:
df.mean(axis = 1)  


0     1.360660
1     1.579077
2     1.472172
3     1.574765
4     1.731241
5     1.557399
6     1.514897
7     1.434554
8     1.651136
9     1.371234
10    1.439188
11    1.483110
12    1.649452
13    1.320386
14    1.632307
15    1.689437
16    1.573717
17    1.520611
18    1.678911
19    1.488426
dtype: float64

In [34]:
print(min(df.mean(axis=0)))
print(df.mean(axis=1).idxmin())

1.4531807558530958
13


/var/folders/7j/pv6r3v3n78bb6v57jglq0z980000gn/T/ipykernel_3320/1881972480.py:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  print(min(df.mean(axis=0)))
/var/folders/7j/pv6r3v3n78bb6v57jglq0z980000gn/T/ipykernel_3320/1881972480.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  print(df.mean(axis=1).idxmin())


In this case we have row 13 which has the lowest mean which is 1.4531807558530958.

<h1>Question11</h1>

In [31]:
df['f']=df['a']+df['b']+df['c']+df['d']+df['e']
df

a         b         c         d         e         f
0   1.880843  1.320479  1.416674  1.005828  1.179477  6.803300
1   1.650108  1.989020  1.484707  1.523982  1.247570  7.895387
2   1.455549  1.440511  1.403900  1.473522  1.587379  7.360861
3   1.851480  1.750142  1.445667  1.253328  1.573210  7.873827
4   1.922452  1.898862  1.697920  1.710930  1.426042  8.656206
5   1.979674  1.068126  1.353029  1.726961  1.659205  7.786996
6   1.507097  1.133749  1.734548  1.663728  1.535361  7.574483
7   1.274441  1.972587  1.701627  1.220980  1.003133  7.172768
8   1.690039  1.568925  1.929697  1.851065  1.215951  8.255678
9   1.474857  1.052607  1.276691  1.635052  1.416964  6.856171
10  1.466008  1.320530  1.511068  1.668908  1.229427  7.195942
11  1.751057  1.475668  1.581855  1.179564  1.427405  7.415549
12  1.865198  1.139561  1.733581  1.905747  1.603171  8.247258
13  1.732365  1.542237  1.128564  1.171042  1.027721  6.601928
14  1.646544  1.799957  1.884981  1.297664  1.532391  8.161537
15  1.941233  1.687333  1.459401  1.433500  1.925718  8.447185
16  1.964113  1.042838  1.978120  1.104549  1.778962  7.868583
17  1.317641  1.323893  1.681040  1.645931  1.634549  7.603055
18  1.362270  1.980773  1.722718  1.355916  1.972880  8.394556
19  1.269388  1.749474  1.739397  1.235419  1.448450  7.442128

In [32]:
df['g']=np.where(df['f']<8,'LT8','GT8')
df

a         b         c         d         e         f    g
0   1.880843  1.320479  1.416674  1.005828  1.179477  6.803300  LT8
1   1.650108  1.989020  1.484707  1.523982  1.247570  7.895387  LT8
2   1.455549  1.440511  1.403900  1.473522  1.587379  7.360861  LT8
3   1.851480  1.750142  1.445667  1.253328  1.573210  7.873827  LT8
4   1.922452  1.898862  1.697920  1.710930  1.426042  8.656206  GT8
5   1.979674  1.068126  1.353029  1.726961  1.659205  7.786996  LT8
6   1.507097  1.133749  1.734548  1.663728  1.535361  7.574483  LT8
7   1.274441  1.972587  1.701627  1.220980  1.003133  7.172768  LT8
8   1.690039  1.568925  1.929697  1.851065  1.215951  8.255678  GT8
9   1.474857  1.052607  1.276691  1.635052  1.416964  6.856171  LT8
10  1.466008  1.320530  1.511068  1.668908  1.229427  7.195942  LT8
11  1.751057  1.475668  1.581855  1.179564  1.427405  7.415549  LT8
12  1.865198  1.139561  1.733581  1.905747  1.603171  8.247258  GT8
13  1.732365  1.542237  1.128564  1.171042  1.027721  6.601928  LT8
14  1.646544  1.799957  1.884981  1.297664  1.532391  8.161537  GT8
15  1.941233  1.687333  1.459401  1.433500  1.925718  8.447185  GT8
16  1.964113  1.042838  1.978120  1.104549  1.778962  7.868583  LT8
17  1.317641  1.323893  1.681040  1.645931  1.634549  7.603055  LT8
18  1.362270  1.980773  1.722718  1.355916  1.972880  8.394556  GT8
19  1.269388  1.749474  1.739397  1.235419  1.448450  7.442128  LT8

Note: since the dataframe has value randomly generated the answers to the questions would vary.